dialvia/http.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 Sauce Labs, Inc. All rights reserved.
+
+package dialvia
+
+import (
+	"bufio"
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+)
+
+type httpProxyDialer struct {
+	dial      ContextDialerFunc
+	proxyURL  *url.URL
+	tlsConfig *tls.Config
+}
+
+func HTTPProxy(dial ContextDialerFunc, proxyURL *url.URL) *httpProxyDialer {
```

Comment:
Exported function returns unexported type.

Reply:
Any alternatives?

Reply:
How about it returns Dialer interface?

Reply:
We could pass the original response in a custom error type.

Reply:
I would rather export it, than wrap with custom error.